### ISSUE TITLE
could you please support ellipse?

### ISSUE THREAD

**Issue by wikywin** [alexbol99/flatten-js#36]

**Comment by dmitru** [comment]
I second that; having ellipse (and perhaps a rectangle) would be very useful!

**Comment by mathedu4all** [comment]
I think we can get ellipse by transform a circle by affine matrix.

**Comment by alexbol99** [comment]
@mathedu4all , transform circle gives ellipse in mathematics, but unfortunately not in this library.
In order to support ellipse I need an object that represent ellipse with two radii rx, ry.  Then I need to implement at least the common methods like distance, intersection, relation and transformation. Not so trivial task.

**Comment by edemaine** [comment]
I'm not sure, but if implementing ellipses, it might make sense to try to support arbitrary [conics](https://en.wikipedia.org/wiki/Conic_section). At least, I think a lot of the math is similar between them, and other conics can be useful to work with. (For example, parabolas come up naturally in some computational geometry like the [medial axis](https://en.wikipedia.org/wiki/Medial_axis).) Even within ellipses, it could be nice to support arbitrary rotation (two foci not necessarily horizontally or vertically aligned). I agree that this is a nontrivial task, but maybe we can aim for it someday. 😅

**Comment by alexbol99** [comment]
I agree, it would be great to support all conic curves, maybe not in the scope of this library.
Even drawing of these curves is not supported neither by SVG nor by Canvas API.
